packages/worker/src/workflow/threaded-vm.ts

```diff
@@ -74,21 +79,25 @@ export class WorkerThreadClient {
 
       completion.resolve(result.output);
     });
-    workerThread.on('exit', () => {
+    workerThread.on('error', (err) => {
+      logger.error(`Workflow Worker Thread failed: ${err}`, err);
+      this.exitError = new UnexpectedError(`Workflow Worker Thread exited prematurely: ${err}`, err);
+      // Node will automatically terminate the Worker Thread, immediately after this event.
+    });
+    workerThread.on('exit', (exitCode) => {
+      logger.trace(`Workflow Worker Thread exited with code ${exitCode}`, { exitError: this.exitError });
       this.workerExited = true;
```

Comment:
Is "exit" guaranteed to be called after "error"? Do we know this.exitError will be set by the error handler first?

Reply:
Node.js' doc is not clear on guarantees here, but it is quite clear from their code that no other event would be received after `exit`, so if we are receive some `error`, that will be before `exit`.

Also, our code kind of deal gracefully with the case where we'd would have unexpectedly received an `exit` event without an `error` first, by just making this an `UnexpectedError` (see below).